recipe/meta.yaml

```diff
@@ -11,7 +11,7 @@ source:
 
 build:
   noarch: python
-  number: 1
+  number: 2
```

Comment:
Command (on windows): `conda search --info unidist-mpi -c conda-forge`
```bash
unidist-mpi 0.4.1 pyhf6b0172_2
------------------------------
file name   : unidist-mpi-0.4.1-pyhf6b0172_2.conda
name        : unidist-mpi
version     : 0.4.1
build       : pyhf6b0172_2
build number: 2
size        : 15 KB
license     : Apache-2.0 AND LicenseRef-Public-Domain
subdir      : noarch
url         : https://conda.anaconda.org/conda-forge/noarch/unidist-mpi-0.4.1-pyhf6b0172_2.conda
md5         : 5e834a417620e87c2f3bd62688c56e93
timestamp   : 2023-09-12 13:49:08 UTC
dependencies:
  - mpi4py >=3.0.3
  - mpich
  - msgpack-python >=1.0.0
  - python >=3.7.1
  - unidist 0.4.1 pyhd8ed1ab_2
```

`mpich` is still among the other deps.


@YarShev looks like conda doesn't publish different packages for different platform. See https://anaconda.org/conda-forge/unidist-mpi/files.

Reply:
We build unidist as a pure python package (with `noarch: python`) so conda only distributes one package for every platform as unidist-mpi is expected to work on win, linux, osx. If I install `conda install -c conda-forge unidist-mpi` on windows, I see the following mpi related packages for installation.

```bash
mpi                conda-forge/win-64::mpi-1.0-msmpi
mpi4py             pkgs/main/win-64::mpi4py-3.1.4-py310hb01be7b_0
msmpi              conda-forge/win-64::msmpi-10.1.1-h3502643_7
unidist            conda-forge/noarch::unidist-0.4.1-pyhd8ed1ab_0
unidist-mpi        conda-forge/noarch::unidist-mpi-0.4.1-pyhd8ed1ab_0
```

We see that unidist-mpi contains the build 0 (latest is the build 2) but the things are installed just fine.

@dalcinl, could we please get your help here in regard of correctness of `conda_build_config.yaml` and `unidist-mpi` output in `meta.yaml`? Is there something we are missing on?

Reply:
@YarShev confirm that now unidist work with python 3.10 https://github.com/modin-project/modin/pull/6548. Thanks!

Reply:
@YarShev I'm not really an expert on conda-build. But you having to explicitly set MPI dependencies seems wrong to me. 

If the `unidist` package is pure-python, then users should be able to use either MPICH and Open MPI as the backend MPI implementation.

For example, if you want to use MPICH

```
conda install unidist-mpi mpich
```

Otherwise, if you want to use Open MPI

```
conda install unidist-mpi openmpi
```

conda's dependency resolver should handle everything.

Reply:
@dalcinl, what you are saying about the conda resolver should work fine but my initial thought was to ship `unidist-mpi` with a concrete MPI implementation to get things working out of the box (like for compatibility). If the user wants to use a specific MPI backend, say Open MPI, he should use the custom-built or system-provided MPI implementation to get ultimate performance. Do you think this would make sense or you are still thinking that it would be better to do what you were saying about? i,e.

```bash
conda install unidist-mpi mpich # or
conda install unidist-mpi openmpi # or
conda install unidist-mpi msmpi
```

Reply:
I understand your motivation, but I you tie the "-mpi" prefix to any particular MPI implementation, that's going to be confusing, because it is not the way other `<pkgname>-mpi` packages in conda-forge work.

>  with a concrete MPI implementation to get things working out of the box (like for compatibility). 

What kind of compatibility are you talking ago.

> If the user wants to use a specific MPI backend, say Open MPI, he should use the custom-built or system-provided MPI implementation to get ultimate performance.

By injecting the "mpich" dependency, you will jeopardize the option of using a system-provided Open MPI.

> Do you think this would make sense or you are still thinking that it would be better to do what you were saying about? i,e.

I would make two builds of `unidist-mpi` that depend on either MPICH or Open MPI, just as mpi4py does. You can make the `unidist-mpi` package that depends on MPICH to have a higher build number, that is, define a "build" variable and multiply the build number by 100. That way if you `conda install unidist-mpi`, the MPI backend you will get by default will be MPICH. Users will still be able to explicitly `conda install unidist-mpi mpich=<spec>`, but they will also be able to `conda install unidist-mpi openmpi=<spec>`, maybe with a external spec to use a system-provided MPI.

Reply:
It would be great if someone would contribute to the conda-forge documentation with these tips about how to support MPI variants in MPI-enabled packages.

Reply:
> What kind of compatibility are you talking ago.

Earlier versions of unidist-mpi contained mpich so I wanted to make it explicit to install unidist-mpi with mpich.

> By injecting the "mpich" dependency, you will jeopardize the option of using a system-provided Open MPI.

I agree, that makes sense.

> I would make two builds of unidist-mpi that depend on either MPICH or Open MPI, just as mpi4py does. You can make the unidist-mpi package that depends on MPICH to have a higher build number, that is, define a "build" variable and multiply the build number by 100. That way if you conda install unidist-mpi, the MPI backend you will get by default will be MPICH. Users will still be able to explicitly conda install unidist-mpi mpich=<spec>, but they will also be able to conda install unidist-mpi openmpi=<spec>, maybe with a external spec to use a system-provided MPI.

I am not sure that I fully get you with respect to two builds. I sketched up a PR https://github.com/conda-forge/unidist-packages-feedstock/pull/12 on this matter. Could you leave your feedbackd there? Is that what you mean?

> It would be great if someone would contribute to the conda-forge documentation with these tips about how to support MPI variants in MPI-enabled packages.

I am happy to contribute the tips to the conda-forge docs when I fully get those tips 😄

Reply:
> Earlier versions of unidist-mpi contained mpich so I wanted to make it explicit to install unidist-mpi with mpich.

OK. Adding a variant that uses Open MPI will not interfere with the MPICH one, so the backward compatibility concern is fully covered.

> I am not sure that I fully get you with respect to two builds. I sketched up a PR #12 on this matter. Could you leave your feedbackd there? Is that what you mean?

I don't think so. I'll drop a comment in the PR.